### PR TITLE
Feature/token auth

### PR DESCRIPTION
**Pull request by toboid**
Once we're happy I'll submit back upstream, to see if can get it in there.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -99,6 +99,21 @@ module.exports = {
                 });
             }
 
+            if (route.security) {
+                var securitySchemes = Object.keys(route.security);
+                Assert.ok(securitySchemes.length < 2, 'HAPI supports 1 authentication strategy per route.');
```
Comment (brainsiq) [inline lib/index.js]: That's not actually true? You can configure multiple strategies through config.auth.strategies.

http://hapijs.com/tutorials/auth#route-configuration

Don't think that's going to cause us any issues but suspect if it was to be merged back into the original module they might want to support more cases/configuration.

**Comment by brainsiq** [comment]
Looks like it will do the job for now! I say for now in case we need scopes later. I guess we will need it to be able to tie indexes to individual clients when searching. That will be fun as the swagger spec says you can only provide scopes for oauth security!

**Comment by toboid** [comment]
Yes, that's right. Swagger spec only allows scopes for OAuth. Think we can handle the relationship between api keys and indexes outside of this mechanism and just specify that a key is required. Doesn't answer the scopes issue when we need that though.

File: lib/index.js (hunk 1 above)
Comment (toboid) [inline lib/index.js]: Ah, didn't see that option. Can add this.

**Comment by toboid** [comment]
Updated to add support for multiple auth strategies per endpoint

**Pull request merged**